tests/kani/Makefile

```diff
@@ -0,0 +1,38 @@
+SHELL=/bin/bash
+
+# Assume cbmc and cbmc-viewer are installed in PATH
+#   skip ./kani/scripts/setup/ubuntu/install_cbmc.sh
+#   skip ./kani/scripts/setup/install_viewer.sh 2.11
+
+KANI=kani
+
+# ARCH is ubuntu or macos-10.15
+ARCH=ubuntu
+
+clone:
+	git clone https://github.com/model-checking/kani.git $(KANI)
+	cd $(KANI) && git submodule update --init
+
+# MacOS warning: If brew fails trying to install ctags over
+# universal-ctags, then replace ctags with universal-ctags
+# ./kani/scripts/setup/macos-10.15/install_deps.sh
+
+build:
+	PATH=$$HOME/.cargo/bin:$$PATH && cd $(KANI) && \
+	    ./scripts/setup/$(ARCH)/install_deps.sh && \
+	    ./scripts/setup/install_rustup.sh && \
+	    cargo build --workspace
+
+test:
+	PATH=$$HOME/.cargo/bin:$$(pwd)/$(KANI)/scripts:$$PATH && \
+	echo cbmc --version: $$(cbmc --version) && \
+	echo cbmc-viewer --version: $$(cbmc-viewer --version) && \
+	echo kani --version: $$(kani --version) && \
+	./write_build_ninja.py $(KANI)/tests/kani && \
+	ninja
+
+clean:
+	$(RM) -r *~ build.ninja .ninja_log __pycache__
```

Comment:
What is `*~` for? Some text editors (emacs) save swap files with this extension, this seems to be trashing those which is unrelated to cleaning out the build? Or does the build itself dump out files like this?

Reply:
The editor emacs makes backup files between saves with names having `~` appended.  I'm going to leave this because it is common cleanup work at the end of hacking.